Fix issues with metric reporting

DanCech opened the pull request:
This PR fixes 2 issues with the current metric reporting framework:

The first commit adds the missing "active users" stat to the data reported to graphite.  This is straightforward and just adds the missing stat.

The second commit fixes an issue that causes the first set of stats reported to graphite after startup to have the system stats all set to 0.  This was a little more involved, since the stats are gathered at startup by `sqlstore.EnsureAdminUser()` but weren't finding their way into the prometheus stats.  Now whenever `GetSystemStats` runs it publishes the stats to the bus and the metrics module registers a listener so it can update the prometheus stats.

DanCech committed:
report active users in graphite stats

DanCech committed:
use bus to publish system stats

torkelo commented on pkg/services/sqlstore/stats.go:
```diff
@@ -68,6 +68,9 @@ func GetSystemStats(query *m.GetSystemStatsQuery) error {
 
 	query.Result = &stats
+
+	bus.Publish(&stats)
```
not sure I understand the publish / even handler addition. What was wrong with using the result property?

torkelo replied:
maybe just calling updateTotalStats after metrics init will fix the initial 0 values and make sure the stats have value before the publisher is started.

DanCech replied:
The reason for doing it this way is so that the prometheus stats can be updated every time we pull the stats from the DB, this means that they're automatically updated at startup by the call we are already making in `EnsureAdminUser` and we don't need to immediately call it again.

More generally this is an implementation of the https://en.wikipedia.org/wiki/Observer_pattern so that we split the responsibility for periodically getting the stats out of the DB from the responsibility for updating the in-memory values, and so that the in-memory values are always up to date regardless of what is triggering the stats to be gathered from the DB.

bergquist replied:
Calling `updateTotalStats()` on  https://github.com/grafana/grafana/blob/master/pkg/metrics/metrics.go#L317 seems like a simpler solution. Since the first thing we do after `EnsureAdminUser` is to set up metrics it shouldn't matter execution time wise. 

I think the Observer pattern is nice when you have multiple sources you want to keep updated. In this case we only have two.

bergquist committed:
metrics: avoid using events unless we have to

this commit also changes the default interval
for updating the stats gauges. Seems like the old values
was a product of previous metrics implementation
